PoseEstimation-TFLiteSwift/PoseConfidenceMapDrawingView.swift

```diff
@@ -0,0 +1,104 @@
+        guard let output = output else { return }
+        
+        let (rowCount, colCount) = (output.dimensions[1], output.dimensions[2])
+        let (width, height) = (frame.width, frame.height)
+        let (oneAreaWidth, oneAreaHeight) = (width/CGFloat(colCount), height/CGFloat(rowCount))
+        
+        // top-left is (0,0)
+        for row in 0..<rowCount {
+            for col in 0..<colCount {
+                var value = outputChannelIndexes.reduce(0.0) { value, outputChannelIndex in
+                    return value + output[0, row, col, outputChannelIndex]
+                }
+                value = min(max(value, -1.0), 1.0)
+                let drawingAreaRect = CGRect(x: oneAreaWidth*CGFloat(col), y: oneAreaHeight*CGFloat(row),
+                                             width: oneAreaWidth, height: oneAreaHeight)
+                drawRect(with: drawingAreaRect, value: value)
+            }
+        }
+    }
+    
+    func drawRect(with rect: CGRect, value: Float32 = 0.0) {
+        guard let startingPoint = rect.points.first else { return }
+        let rectPath = UIBezierPath()
+        rectPath.move(to: startingPoint)
+        for point in rect.points[1...] {
+            rectPath.addLine(to: point)
+        }
+        rectPath.addLine(to: startingPoint)
+        rectPath.close()
+
+        rectPath.lineWidth = DrawingConstant.Line.width
+        DrawingConstant.Area.lineColor.setStroke()
+        rectPath.stroke()
+        DrawingConstant.Area.areaColor(CGFloat(value)).setFill()
+        rectPath.fill()
+    }
+}
+
+private extension PoseConfidenceMapDrawingView {
+    enum DrawingConstant {
+        enum Line {
+            static let width: CGFloat = 0.5
+            static let color: UIColor = UIColor(red: 0.5, green: 0.1, blue: 0.1, alpha: 1)
+        }
+        enum Area {
+            static let lineColor: UIColor = UIColor(red: 1, green: 1, blue: 1, alpha: 0.3)
+            static func areaColor(_ value: CGFloat) -> UIColor {
+                if value < 0 {
+                    return areaNegativeColor(value)
+                } else {
+                    return areaPositiveColor(value)
+                }
+            }
+            private static let baisAlpha: CGFloat = 0.5 // 0.85
+            private static func areaPositiveColor(_ value: CGFloat) -> UIColor {
+                let colorValue = min(max(value, 0.0), 1.0)
+                let alphaValue = (1-baisAlpha)*colorValue + baisAlpha
+                return UIColor(red: colorValue, green: 0, blue: 0, alpha: alphaValue)
+            }
+            private static func areaNegativeColor(_ value: CGFloat) -> UIColor {
+                let colorValue = min(max(abs(value), 0.0), 1.0)
+                let alphaValue = (1-baisAlpha)*colorValue + baisAlpha
+                return UIColor(red: 0, green: colorValue, blue: 0, alpha: alphaValue)
+            }
+        }
+    }
+}
+
+private extension CGPoint {
+    func scaled(to ratioSize: CGSize) -> CGPoint {
```

Comment:
I think,
This function's behavior mean 'coordinate move'.

Reply:
The term `scale` was used in affine transformation.
 I think `coordinate move` corresponds to `transform`.
![image](https://user-images.githubusercontent.com/37643248/80665820-73947e00-8ad5-11ea-97cf-6609bd95ba66.png)
 ref: https://neutrium.net/mathematics/basics-of-affine-transformation/

Reply:
👍 
Good,
I got the fact.
thx.